core/client/fs/src/main/java/alluxio/client/file/ufs/UfsBaseFileSystem.java

```diff
@@ -294,8 +295,13 @@ public FileInStream openFile(AlluxioURI path, OpenFilePOptions options) {
   public FileInStream openFile(URIStatus status, OpenFilePOptions options) {
     return callWithReturn(() -> {
       // TODO(lu) deal with other options e.g. maxUfsReadConcurrency
-      return new UfsFileInStream(mUfs.get().open(status.getPath()),
-          status.getLength());
+      return new UfsFileInStream(offset -> {
+        try {
+          return mUfs.get().open(status.getPath(), OpenOptions.defaults().setOffset(offset));
```

Comment:
Ideally, all UFS streams can support seek() or just throw exceptions if not supported. But Fuse heavily rely on seek() so if a UFS input stream does not support seek(), it basically cannot be supported by FUSE.
As a workaround, similar to what Alluxio does today, when seek happens, close the current stream and reopen a new one with a new offset.

Reply:
Agree that it's not safe to use AlluxioRuntimeException in UFS because it could be used in other places, e.g. journal

Can we use it in this class which is only used by FUSE, and FUSE has already changed to use AlluxioRuntimeException

Reply:
This class has already change to throw RuntimeException in 
```
  private static void call(UfsCallable callable) {
    try {
      callable.call();
    } catch (IOException e) {
      throw AlluxioRuntimeException.from(e);
    }
  }

```